fl4health/model_bases/autoencoders_base.py

```diff
@@ -0,0 +1,144 @@
+        mu, logvar = self.encoder(input)
+        return mu, logvar
+
+    def decode(self, latent_vector: torch.Tensor) -> torch.Tensor:
+        output = self.decoder(latent_vector)
+        return output
+
+    def sampling(self, mu: torch.Tensor, logvar: torch.Tensor) -> torch.Tensor:
+        std = torch.exp(0.5 * logvar)
+        eps = torch.randn_like(std)
+        return eps.mul(std).add_(mu)
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        mu, logvar = self.encode(input)
+        z = self.sampling(mu, logvar)
+        output = self.decode(z)
+        return torch.cat((logvar, mu, output), dim=1)
+
+
+class ConditionalVAE(AutoEncoderBase):
+    def __init__(
+        self,
+        model_type: AutoEncoderType,
+        num_conditions: int,
+        encoder: nn.Module,
+        decoder: nn.Module,
+    ) -> None:
+        """Initializes the ConditionalVAE class.
+
+        Args:
+            model_type (AutoEncoderType): The model type should be CONDITIONAL_VAE.
+            num_conditions (int): The total number of conditions, which is used to prepare the condition vector.
+            encoder (nn.Module): Encoder module defined by the user.
+            decoder (nn.Module): Decoder module defined by the user.
+        """
+        super().__init__(model_type, encoder, decoder)
+        self.num_conditions = num_conditions
+
+    def cat_input_condition(self, input: torch.Tensor, condition: torch.Tensor) -> torch.Tensor:
+        input_cond = torch.cat((input, condition), dim=-1)
+        return input_cond
+
+    def one_hot(self, condition: torch.Tensor) -> torch.Tensor:
+        # One-hot encoding of the condition
+        condition = F.one_hot(condition.to(torch.int64), self.num_conditions).to(condition.device)
+        return condition
+
+    def encode(self, input: torch.Tensor, condition: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        input = self.cat_input_condition(input, condition)
+        mu, logvar = self.encoder(input)
+        return mu, logvar
+
+    def decode(self, latent_vector: torch.Tensor, condition: torch.Tensor) -> torch.Tensor:
+        latent_vector = self.cat_input_condition(latent_vector, condition)
+        output = self.decoder(latent_vector)
+        return output
+
+    def sampling(self, mu: torch.Tensor, logvar: torch.Tensor) -> torch.Tensor:
+        std = torch.exp(0.5 * logvar)
+        eps = torch.randn_like(std)
+        return eps.mul(std).add_(mu)
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        x = input[:, :-1]  # Exclude the last column (conditions)
+        condition = input[:, -1]  # Last column contains conditions
+        condition = self.one_hot(condition)
+        mu, logvar = self.encode(x, condition)
+        z = self.sampling(mu, logvar)
+        output = self.decode(z, condition)
+        return torch.cat((logvar, mu, output), dim=1)
+
+
+class Basic_AE(AutoEncoderBase):
```

Comment:
Following the definitions of other AE you have defined, maybe BasicAE is a more suitable name?

Reply:
I agree!